TMGToolbox/src/assignment/preprocessing/remove_redundant_functions.py

```diff
@@ -0,0 +1,296 @@
+'''
+    Copyright 2019 Travel Modelling Group, Department of Civil Engineering, University of Toronto
+
+    This file is part of the TMG Toolbox.
+
+    The TMG Toolbox is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The TMG Toolbox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with the TMG Toolbox.  If not, see <http://www.gnu.org/licenses/>.
+'''
+
+#---METADATA---------------------
+'''
+Remove Redundant Functions
+
+    Authors: Luna Xi
+
+    Latest revision by: 
+    
+    [Description]
+    This tool is used to compare the functions in the networks and the database, 
+    it will export all redundant functions (i.e., not used in any scenario) as an individual file 
+    and remove them from the database.
+        
+'''
+#---VERSION HISTORY
+'''
+    0.0.1 Created
+    
+'''
+
+import inro.modeller as _m
+import traceback as _traceback
+from contextlib import contextmanager
+from contextlib import nested
+_M = _m.Modeller() 
+_exportFcns = _M.tool('inro.emme.data.function.export_functions')
+_deleteFcns = _M.tool('inro.emme.data.function.delete_function')
+_util = _M.module('tmg.common.utilities')
+_tmgTPB = _M.module('tmg.common.TMG_tool_page_builder')
+
+##########################################################################################################
+
+class RemoveRedundantFunctions(_m.Tool()):
```

Comment:
It seems like this tool is actually trying to remove unused functions, not redundant functions.